src/test/java/uk/gov/hmcts/reform/bailcaseapi/domain/handlers/presubmit/ChangeDirectionDueDateHandlerTest.java

```diff
@@ -0,0 +1,281 @@
+        assertEquals(dateSent.toString(), actualDirections.get(1).getValue().getDateSent());
+
+        assertEquals(1, actualDirections.get(1).getValue().getPreviousDates().size());
+        assertEquals("1", actualDirections.get(1).getValue().getPreviousDates().get(0).getId());
+        assertEquals("2020-11-01",
+            actualDirections.get(1).getValue().getPreviousDates().get(0).getValue().getDateDue());
+        assertEquals("2019-11-01",
+            actualDirections.get(1).getValue().getPreviousDates().get(0).getValue().getDateSent());
+    }
+
+    @Test
+    void handling_should_throw_if_cannot_actually_handle() {
+
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.handle(PreSubmitCallbackStage.ABOUT_TO_START, callback))
+            .hasMessage("Cannot handle callback")
+            .isExactlyInstanceOf(IllegalStateException.class);
+
+        when(callback.getEvent()).thenReturn(Event.SEND_BAIL_DIRECTION);
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, callback))
+            .hasMessage("Cannot handle callback")
+            .isExactlyInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    void it_can_handle_callback() {
+
+        for (Event event : Event.values()) {
+
+            when(callback.getEvent()).thenReturn(event);
+
+            for (PreSubmitCallbackStage callbackStage : PreSubmitCallbackStage.values()) {
+
+                boolean canHandle = changeDirectionDueDateHandler.canHandle(callbackStage, callback);
+
+                if (event == Event.CHANGE_BAIL_DIRECTION_DUE_DATE
+                    && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT) {
+
+                    assertTrue(canHandle);
+                } else {
+                    assertFalse(canHandle);
+                }
+            }
+
+            reset(callback);
+        }
+    }
+
+    @Test
+    void should_not_allow_null_arguments() {
+
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.canHandle(null, callback))
+            .hasMessage("callbackStage must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.canHandle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, null))
+            .hasMessage("callback must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.handle(null, callback))
+            .hasMessage("callbackStage must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+
+        assertThatThrownBy(() -> changeDirectionDueDateHandler.handle(PreSubmitCallbackStage.ABOUT_TO_SUBMIT, null))
+            .hasMessage("callback must not be null")
+            .isExactlyInstanceOf(NullPointerException.class);
+    }
+
+    // remove when new CCD definitions are in Prod
```

Comment:
Please check the classes and tests. We don't have any outdated ccd definitions and we should be working on what works with our set of definitions.